davidfoo07 iP

davidfoo07 opened the pull request:
# **G**
>## “Your mind is for having ideas, not holding them.” – David Allen (source)

G frees your mind from remembering tasks. It’s:

-   text-based
-   easy to learn
-   **_FAST, SUPER FAST_** to use!

All you need to do is:

1. Clone the repository with this command: git clone https://github.com/davidfoo07/ip.git
2. cd into the ip directory.
3. Run it with this command: java -jar build/libs/g.jar
4. Add your tasks.
5. Let it manage your tasks for you 😉

And the best part? It’s **FREE**!

Features:

-   [x] Managing tasks
-   [ ] Managing deadlines (coming soon)
-   [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

TechMyUnilah committed:
rename, greet and exit

TechMyUnilah committed:
edit the message

TechMyUnilah committed:
try to create tag 0

TechMyUnilah committed:
trying on tag

TechMyUnilah committed:
echo

TechMyUnilah committed:
level2. add, list

TechMyUnilah committed:
Level-3: Mark As Done

TechMyUnilah committed:
remodified Level-2 to display list is empty

TechMyUnilah committed:
Level 4: todo, event and deadline

TechMyUnilah committed:
commit

TechMyUnilah committed:
text-ui-test

TechMyUnilah committed:
Merge branch 'l4'

TechMyUnilah committed:
Level-6: delete

TechMyUnilah committed:
level 5

TechMyUnilah committed:
level 6 again

TechMyUnilah committed:
Level7: Save

zacklow28 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+public class Deadline extends Task {
+
+    protected String by;
```
the naming of by can be changed to date
cause by can be "by John"

twhjames replied:
I agree with this suggestion. Renaming `by` to `date` would make the purpose of the variable clearer 👍

zacklow28 commented on src/main/java/G.java:
```diff
@@ -0,0 +1,166 @@
+
+
+    // ✅ Print task list
+    private static void printTaskList(ArrayList<Task> tasks) {
```
good method names here, clear and describe the method

zacklow28 commented on src/main/java/G.java:
```diff
@@ -0,0 +1,166 @@
+
+            // Read file content if it exists
+            BufferedReader br = new BufferedReader(new FileReader(file));
+            String line;
```
what is line here? Can be more specific 😃

twhjames replied:
I like the descriptive comments as it make the code easier to understand, building onto @zacklow28 suggestion,  you might consider renaming the `line` variable to something more specific, such as `taskEntry` or `fileLine`, to better reflect its purpose

looking at the `saveTasksToFile()` method,`taskEntry` might be a more consistent choice for maintaining clarity across both methods

zacklow28 reviewed (commented):
LGTM, most naming conventions are followed. Just some nits

twhjames commented on src/main/java/G.java:
```diff
@@ -0,0 +1,166 @@
+
+        ArrayList<Task> tasks = loadTasksFromFile(); // Load tasks from file
+        Scanner scn = new Scanner(System.in);
+        String str1 = scn.nextLine().trim();
```
Consider renaming `str1` to a more descriptive variable name such as `userInput`.
```suggestion
        String userInput = scn.nextLine().trim();
```

twhjames commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public Task(String description, Boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
```
Consider adding descriptive header comments for public classes and methods. You can refer to [this](https://se-education.org/guides/conventions/java/intermediate.html#comments) section of Java Coding Standard.

twhjames commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+        this.isDone = isDone;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]"); // mark done task with X
+    }
```
This is a well-named method 👍

TechMyUnilah committed:
Level 8:  Date and times

TechMyUnilah committed:
Done with A-MoreOOP and proceeding with A-Gradle

TechMyUnilah committed:
Merge branch 'add-gradle-support'

TechMyUnilah committed:
A-gradle and A-JUnitTest

TechMyUnilah committed:
written java doc

TechMyUnilah committed:
commit

TechMyUnilah committed:
Level-9 : Find

TechMyUnilah committed:
Merge branch 'branch-A-JavaDoc'

TechMyUnilah committed:
Merge branch 'branch-Level-9'

TechMyUnilah committed:
Level-10

davidfoo07 committed:
Update README.md

TechMyUnilah committed:
Refactor Core Chatbot Components for Improved Stability, Input Validation, and Error Handling

## Summary
This commit refactors multiple core components (`Parser.java`, `Duke.java`, `AddCommand.java`, `MainWindow.java`, `TaskList.java`, and `Ui.java`) to improve stability, prevent crashes, and enhance error handling. Assertions have been added to enforce assumptions, and meaningful error messages are now provided for incorrect inputs.

## Changes Made:

### 🔹 **Parser.java**
- **Enhanced Input Validation:**
  - Prevents `ArrayIndexOutOfBoundsException` when commands lack arguments.
  - Ensures `"todo"` has a description.
  - Validates `"deadline"` and `"event"` formats, checking for `/by`, `/from`, and `/to`.
  - Ensures `"mark"`, `"unmark"`, and `"delete"` commands have numeric indices.
  - Added helper method `isValidIndex()` to validate task indices.

- **Improved Error Handling:**
  - Returns `InvalidCommand` with meaningful error messages instead of crashing.
  - Example: `"Usage: deadline <task> /by <date>"` instead of an exception.

- **Added Assertions:**
  - Ensures `userInput` is not null or empty.
  - Validates parsed arguments before processing.

### 🔹 **Duke.java**
- **Added Assertions:**
  - Ensure `ui`, `storage`, and `tasks` are initialized correctly in the constructor.
- **Updated `getResponse()` to prevent null responses:**
  - Ensures chatbot always returns a valid string.
  - Automatically saves after every command execution.

### 🔹 **AddCommand.java**
- **Added Assertions:**
  - Ensures a task is valid before adding it to the task list.
- **Improved Error Handling:**
  - Prevents adding `null` tasks.

### 🔹 **MainWindow.java**
- **Ensured GUI Components Are Properly Loaded:**
  - Added assertions in `initialize()` to ensure `scrollPane`, `dialogContainer`, `userInput`, and `sendButton` are not null.
- **Handled Exit Command Properly:**
  - Calls `Platform.exit()` when the user types `"bye"`.

### 🔹 **TaskList.java**
- **Prevented Invalid Task Index Errors:**
  - Added assertions to check valid indices in `deleteTask()`, `markTask()`, and `unmarkTask()`.
  - Ensures task operations do not access out-of-bounds indices.

### 🔹 **Ui.java**
- **Ensured UI Messages Are Not Null:**
  - Added assertions to `showMessage()`.
- **Improved Error Messages:**
  - Returns more helpful responses for invalid commands.

## Impact:
- **Prevents chatbot crashes** due to malformed input.
- **Provides users with clear error messages** instead of silent failures or exceptions.
- **Ensures stability** by enforcing assumptions via assertions.
- **Improves maintainability** with better structured validation and error handling.

This refactor significantly enhances the robustness of the chatbot, ensuring a smoother user experience and reducing unexpected runtime errors. 🚀

TechMyUnilah committed:
Merge branch 'master' of https://github.com/davidfoo07/ip

TechMyUnilah committed:
Merge branch 'Branch-A-Assertions'

TechMyUnilah committed:
Enhance Duplicate Task Detection in TaskList to Prevent Redundant Entries

## Summary
This commit improves task management by adding **duplicate task detection**, ensuring that the chatbot prevents users from adding the same task multiple times. The changes affect `TaskList.java`, `Task.java`, and `AddCommand.java`, making task comparison more robust and enforcing validation before adding a new task.

## Changes Made:

### 🔹 **TaskList.java**
- **Added `isDuplicate()` method** to check if a task with the same description already exists.
- **Updated `addTask()`** to prevent duplicate tasks from being added.
- **Returns a user-friendly message** when a duplicate task is detected.

### 🔹 **Task.java**
- **Overrode `equals()` method** to compare tasks based on their **description** instead of object reference.
- **Case-insensitive comparison** ensures `"Read book"` and `"read book"` are treated as the same task.
- **Prevents duplicate tasks from being added even if they are separate instances** in memory.

### 🔹 **AddCommand.java**
- **Modified `execute()` to use `TaskList.addTask()`**.
- **Ensures duplicate tasks are not added** by relying on the updated `TaskList` logic.
- **Improved validation** before adding a task.

## Impact:
- **Prevents duplicate tasks** from cluttering the task list.
- **Improves user experience** by providing a warning when a duplicate is detected.
- **Enhances task consistency** by ensuring each task in the list is unique.
- **Makes task comparison more efficient** through the `equals()` override.

This enhancement strengthens the chatbot's task management, making it more **intelligent and user-friendly**. 🚀

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

davidfoo07 committed:
Update README.md

TechMyUnilah committed:
written java doc in each file and updated the readme.md in docs

TechMyUnilah committed:
Merge branch 'master' of https://github.com/davidfoo07/ip

TechMyUnilah committed:
Included Ui.png in readme

davidfoo07 committed:
Update README.md

TechMyUnilah committed:
updated readMe

TechMyUnilah committed:
Merge branch 'master' of https://github.com/davidfoo07/ip

TechMyUnilah committed:
Merge commit 'ff689c9df97a6a1924995d0ff5d4982da9421a47'

TechMyUnilah committed:
renamed duke into G

TechMyUnilah committed:
Merge branch 'master' into branch-A-Assertions

davidfoo07 committed:
Merge pull request #2 from davidfoo07/branch-A-Assertions

renamed duke into G

TechMyUnilah committed:
renamed duke to g

TechMyUnilah committed:
updated duke references in these files to g

TechMyUnilah committed:
Merge branch 'master' into branch-A-CodeQuality

davidfoo07 committed:
Merge pull request #3 from davidfoo07/branch-A-CodeQuality

### **Renamed Duke to G & Updated Dependencies** 🚀  

#### **Summary**  
This pull request renames all references of **Duke** to **G** across the project and updates necessary configurations to ensure smooth functionality.

#### **Changes Made**  
✅ **Codebase Refactor:**  
- **Renamed the `duke` folder** to `g` for consistency.  
- **Renamed `Duke.java` to `G.java`** to align with the new project name.  
- **Updated `package` declarations** from `package duke;` to `package g;`.  
- **Modified all `import` statements** referencing `duke` to use `g` instead.  

✅ **GUI Updates:**  
- **Updated `MainWindow.java`** to reference `g` instead of `duke`.  
- **Refactored FXML files (`MainWindow.fxml`)** to replace all occurrences of `duke` with `g`.  

✅ **Build System Updates:**  
- **Modified `build.gradle` application config**:  
  - Changed `application { mainClass.set("duke.gui.Launcher") }` → `application { mainClass.set("g.gui.Launcher") }`.  
  - Ensured all **Gradle dependencies** referencing `duke` were replaced with `g`.  

#### **Impact**  
- Maintains **consistent project naming** across all files.  
- Ensures **GUI components properly reference `G` instead of `Duke`**.  
- **Build process remains functional** with updated `build.gradle` configuration.  

#### **Verification**  
- **Code compiles successfully** after renaming.  
- **GUI runs properly** without breaking references.  
- **No unresolved imports or package issues** remain.

TechMyUnilah committed:
### **Commit Message:** 🧹 **Removed Unused `Duke.class` File**

---

#### **Summary:**
- **Deleted** the **outdated** and **unused** `Duke.class` file from the **source directory**.
- This **cleanup** is part of the **refactoring effort** to **rename Duke to G**, ensuring no **legacy files** remain in the **codebase**.

---

#### **Reason for Removal:**
- The `Duke.class` file was **no longer in use**, as **all references** to `Duke` have been **updated** to `G`.
- Keeping **obsolete class files** may **cause confusion** and **potential conflicts** during **compilation** and **execution**.
- **Prevents** any **accidental usage** of the **old class file**.

---

#### **Impact:**
- **Improves code cleanliness** by **removing unnecessary files**.
- **Reduces potential build issues**, especially with **auto-compiling tools** like **Gradle**.
- Helps **maintain a clear project structure**, making the **repository easier to navigate**.

---

#### **Verification:**
- **Ran the build process** to ensure there are **no broken references**.
- **Tested the application**, confirming that **all functionalities work correctly** without the **`Duke.class`** file.

---

#### **Next Steps:**
- **Ensure no other legacy files remain** in the project.
- Conduct a **final sweep of the repository** to **identify any remnants of the `Duke` naming**.
- **Continue testing** to verify **smooth functionality** throughout the **refactoring process**.

---

This **commit** is a **small yet important step** in **maintaining the project's quality and consistency**, ensuring a **clean and professional codebase**. 😊

TechMyUnilah committed:
### **Commit Message:** 🧹 **Removed Outdated `Parser.class` File & Switched to Gradle Build**

---

#### **Summary:**
- **Deleted** the **obsolete** `Parser.class` file, as the **project now uses Gradle** for **building and compiling** instead of a **manual decompiler**.

---

#### **Changes Made:**
- **Removed `Parser.class`** from the **source directory**, eliminating **unnecessary compiled files**.
- Updated the **build process** to use **Gradle** exclusively, ensuring **consistent and automated builds**.

---

#### **Reason for Removal:**
- The `Parser.class` file was a **legacy artifact** from using a **manual decompiler (FernFlower)**.
- **Switching to Gradle** provides a **cleaner and more efficient build process**, **automating compilation** and **dependency management**.
- **Prevents potential conflicts** and **outdated references** caused by **manually handled class files**.

---

#### **Impact:**
- **Improves build consistency** by **relying on Gradle** for **class file generation**.
- **Reduces manual overhead**, as **Gradle** handles **compilation, testing, and packaging**.
- Ensures a **cleaner project structure**, making the **codebase easier to manage and maintain**.

---

#### **Verification:**
- **Cleaned the project** using:
```sh
./gradlew clean
```
- **Rebuilt the project** with:
```sh
./gradlew build
```
- **Tested the application**, confirming that **all functionalities work correctly** without the **`Parser.class`** file.

---

#### **Next Steps:**
- Continue using **Gradle** for **future builds** to **maintain a clean and consistent build environment**.
- **Remove any remaining manually compiled `.class` files**, ensuring the **project relies solely on Gradle builds**.
- Perform **smoke tests** to verify that the **application runs correctly** on **all supported platforms**.

---

This **commit** is part of the **ongoing effort** to **modernize the build process**, ensuring **high code quality**, **ease of maintenance**, and **smooth deployment**. 😊

TechMyUnilah committed:
Removed the obsolete `Storage.class` file from the source directory.

TechMyUnilah committed:
Revert "### **Commit Message:** 🧹 **Removed Outdated `Parser.class` File & Switched to Gradle Build**"

This reverts commit 3f312995a20a4fcd4a8a73468284a8a94137834b.

TechMyUnilah committed:
Revert "### **Commit Message:** 🧹 **Removed Unused `Duke.class` File**"

This reverts commit 1ab3703360bc6088763a91de2b54f7307c11c7a0.

TechMyUnilah committed:
Removed the obsolete Storage.class file from the source directory.

TechMyUnilah committed:
Removed the obsolete Parser.class file from the source directory.

TechMyUnilah committed:
Revert "Removed the obsolete Storage.class file from the source directory."

This reverts commit 5283e73e133e01a4ba0585e7f27f75f45a42c89e.

TechMyUnilah committed:
Removed the obsolete Duke.class file from the source directory.

TechMyUnilah committed:
Removed the obsolete Deadline.class file from the source directory.

TechMyUnilah committed:
Removed the obsolete Event.class file from the source directory.

TechMyUnilah committed:
Merge branch 'branch-A-CodeQuality'

TechMyUnilah committed:
Merge branch 'branch-A-CodeQuality'

TechMyUnilah committed:
Removed the obsolete Command.class file from the source directory

TechMyUnilah committed:
Removed the obsolete DeleteCommand.class file from the source directory

TechMyUnilah committed:
Removed the obsolete ExitCommand.class file from the source directory

TechMyUnilah committed:
Removed the obsolete InvalidCommand.class
file from the source directory

TechMyUnilah committed:
Remove the obsolete ListCommand.class

TechMyUnilah committed:
Remove the obsolete MarkCommand.class

TechMyUnilah committed:
Remove the obsolete UnmarkCommand.class

TechMyUnilah committed:
Remove the obsolete TaskCommand.class

TechMyUnilah committed:
Remove the obsolete TaskList.class
